src/library/sidebarmodel.cpp

```diff
@@ -112,26 +112,29 @@ QModelIndex SidebarModel::index(int row, int column,
         if (parent.internalPointer() == this) {
             const QAbstractItemModel* childModel = m_sFeatures[parent.row()]->getChildModel();
             QModelIndex childIndex = childModel->index(row, column);
-            TreeItem* tree_item = (TreeItem*)childIndex.internalPointer();
-            if (tree_item && childIndex.isValid()) {
-                return createIndex(childIndex.row(), childIndex.column(), (void*)tree_item);
+            TreeItem* pTreeItem = static_cast<TreeItem*>(childIndex.internalPointer());
+            if (pTreeItem && childIndex.isValid()) {
+                return createIndex(childIndex.row(), childIndex.column(), pTreeItem);
             } else {
                 return QModelIndex();
             }
         } else {
             // We have selected an item within the childmodel
             // This item has always an internal pointer of (sub)type TreeItem
-            TreeItem* tree_item = (TreeItem*)parent.internalPointer();
-            if (row < tree_item->childRows()) {
-                return createIndex(row, column, (void*) tree_item->child(row));
+            TreeItem* pTreeItem = static_cast<TreeItem*>(parent.internalPointer());
+            if (row < pTreeItem->childRows()) {
+                return createIndex(row, column, pTreeItem->child(row));
             } else {
                 // Otherwise this row might have been removed just now
                 // (just a dirty workaround for unmaintainable GUI code)
                 return QModelIndex();
             }
         }
     }
-    return createIndex(row, column, (void*)this);
+
+    // `this` is const, but the function expects a non-const pointer.
+    // TODO: Check if we can get rid of this const cast somehow.
+    return createIndex(row, column, const_cast<SidebarModel*>(this));
```

Comment:
This is IMHO one of the rare cases where a const cast is correct. 
The function itself is const, and it returns an object with a pointer to itself. 
Nothing changes in the object before the function is done. So the const nature is maintained.  

Interestingly this function takes a const pointer in Qt 6 here: 
https://github.com/qt/qtbase/blob/9db7cc79a26ced4997277b5c206ca15949133240/src/corelib/itemmodels/qabstractitemmodel.h#L404

I would prefer to cast to the parameter type directly:```suggestion
    return createIndex(row, column, const_cast<void*>(this));
```

Reply:
Taking the this pointer is redundant by the way: 
https://github.com/qt/qtbase/blob/9db7cc79a26ced4997277b5c206ca15949133240/src/corelib/itemmodels/qabstractitemmodel.h#L457

We can access it by QModelIndex::model()

However there is a warning 

> A const pointer to the model is returned because calls to non-const functions of the model might invalidate the model index and possibly crash your application.

That is what we do now ... :-/

https://doc.qt.io/qt-6/qmodelindex.html#model

Reply:
Do you have interest to check, if we can just replace internalPointer() by model() and keep the constnes of the pointer?

Reply:
> Do you have interest to check, if we can just replace internalPointer() by model() and keep the constnes of the pointer?

I don't really get what you mean. Can you give me a code example?